### ISSUE TITLE
Training problem

### ISSUE THREAD

**Issue by mgqwyz** [Zengyi-Qin/MonoGRNet#12]
Thanks for your great work.
Some training problems really confused me.

I tried to train the Detector without the pretrained model_3D and I use the pretrained model_2D.
I set leaning rate 1e-4 and joint_3D = True for 80K iteration. The 3D detection eval result is  

 0.410524 0.319438 0.272677.

And the I set leaning rate 1e-5 using SGD and joint_2D_3D = True for 40K iteration but the result going worse.

I wonder how to set the hyper parameter during the training process during the second and the third training stage to reach your wonderful performance

**Comment by Zengyi-Qin** [comment]
Thank you for your interest!
There're actually a lot of ways to reach the reported performance without the pretrained 3D model.
Please use the following config that is simplier than described in the paper:
(1) Set joint_2D_3D = True and others=False in both the optimizer/generic_optimzer.py and decoder/fastBox.py
(2) Fix lr=1e-5 and always use Adam.
In this way the model is supposed to reach 36+ AP@IoU=0.5, moderate, at step 120000~160000.

**Comment by Zengyi-Qin** [comment]
And reduce the evaluation intervals (in ./hypes/). Usually when the training step has exceeded 120000, you should use a smaller evaluation interval (about 500, or 800 steps) in case you miss the best performace.

**Comment by mgqwyz** [comment]
Thank you for your reply.
I tried your training method, when running the third Evaluation part, memory(32GB) ran out.
So I am trying evaluate every model between 120000~160000, it takes a long time.
I hope to reach 36AP :)

**Comment by Zengyi-Qin** [comment]
Other people also encountered memory problems in evaluation.
Please run `make` in `./submodules/kittiEvaluation` to recompile the KITTI evaluation script.

During evaluation, the python script calls `./submodules/kittiEvaluation/evaluate_object` as a subprocess. Check `./evals/kitti_eval.py` for details.

So another solution may be killing the previous subprocess, cuz every time the evaluation script `./evals/kitti_eval.py`  will call a new subprocess. But anyway I didn't encounter this problem, and I don't need to kill those processes mannually. So pls try to recompile the script as described above.

**Comment by mgqwyz** [comment]
thanks a lot.

I have found that when running the func evaluation in kittieval.py , the sess.run() takes many memory(RAM). I tried your method but It doesn’t work, maybe it’s because sth wrong in inference.

I also found that your model have a high training score close to 90 ,  Do I need to get the same result?
I set batchsize=4 and training 300000 , it still cannot get an imaginable result. The training score is just over 50, and the lower total loss is around 0.5 instead of 0.1 , maybe I need to train the network using other flags.

**Comment by Zengyi-Qin** [comment]
There're many output scores. Could you let me know which score is around 50? The 3D detection AP at IoU=0.5 should be around 36 for moderate. The evaluation is (typically) done under IoU threshold of 0.5, you can get a much higher score if lowering the threshold.

**Comment by mgqwyz** [comment]
Sorry, I fail expressing the problem, the evaluation  threshold is 0.5 and the score which I say is the 3D detection moderate AP of the train_out instead of val_out. And the best val detection AP at IOU=0.5 is 32 for moderate.
